generated/appplatform/microsoft.appplatform/2022-01-01-preview/types.md

```diff
@@ -835,19 +839,22 @@
 ### JarUploadedUserSourceInfo
 #### Properties
 * **jvmOptions**: string: JVM parameter
+* **relativePath**: string: Relative path of the storage which stores the source
 * **runtimeVersion**: string: Runtime version of the Jar file
 * **type**: 'Jar' (Required): Type of the source uploaded
 
 ### NetCoreZipUploadedUserSourceInfo
 #### Properties
 * **netCoreMainEntryPath**: string: The path to the .NET executable relative to zip root
+* **relativePath**: string: Relative path of the storage which stores the source
 * **runtimeVersion**: string: Runtime version of the .Net file
 * **type**: 'NetCoreZip' (Required): Type of the source uploaded
 
 ### SourceUploadedUserSourceInfo
 #### Properties
 * **artifactSelector**: string: Selector for the artifact to be used for the deployment for multi-module projects. This should be
 the relative path to the target module/project.
+* **relativePath**: string: Relative path of the storage which stores the source
```

Comment:
This change is correct.

`JarUploadedUserSourceInfo`, `NetCoreZipUploadedUserSourceInfo`, and `SourceUploadedUserSourceInfo` are not immediate descendants of the enum base shape. The intermediate ancestor`UploadedUserSourceInfo` defines a `relativePath` property and is the immediate parent of the three aforementioned branches and the direct child of the enum base shape. Cf https://github.com/Azure/azure-rest-api-specs/blob/37c9b56303da9ec425b00575c0027b44d11a772f/specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2022-01-01-preview/appplatform.json#L8555-L8595